src/service/extensions-impl.js

```diff
@@ -572,6 +572,14 @@ export class Extensions {
     scriptElement.setAttribute('data-script', extensionId);
     scriptElement.setAttribute('i-amphtml-inserted', '');
 
+    // Propagate nonce to all generated script tags.
+    if (this.win.document.currentScript.hasAttribute('nonce')) {
```

Comment:
From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript):

> It's important to note that this will not reference the <script> element if the code in the script is being called as a callback or event handler; it will only reference the element while it's initially being processed.

This is always being called from a callback.